File: Server/src/MeleeManager.ts
import { ITemplateItem } from "@spt-aki/models/eft/common/tables/ITemplateItem"
import { AbstractModManager } from "./AbstractModManager"
import { Constants } from "./Constants"
import { Helper } from "./Helper"


export class MeleeManager extends AbstractModManager
{
    protected configName: string = "MeleeConfig"

    protected afterPostDB(): void
    {
        Helper.iterateConfigItems(this.databaseTables, this.config, this.setMeleeItem)
        
        console.log(`${Constants.ModTitle}: Melee changes applied!`)
    }
    
    private setMeleeItem(item: ITemplateItem, config: any)
    {
        item._props.knifeHitSlashDam = config.slashDamage
        item._props.knifeHitStabDam = config.stabDamage
    }
}

File: Server/VAI-NRTP/Managers/MeleeManager.cs
using System.Text.Json.Nodes;
using SPTarkov.DI.Annotations;
using SPTarkov.Server.Core.Models.Eft.Common.Tables;

namespace VAI.NRTP.Managers;

[Injectable(InjectionType.Singleton)]
public sealed class MeleeManager : AbstractModManager
{
    protected override string ConfigName => "MeleeConfig";

    protected override void AfterPostDb()
    {
        Helper.IterateConfigItems(DatabaseTables, Config, SetMeleeItem);

        Constants.GetLogger().Info($"{Constants.ModTitle}: Melee changes applied!");
    }

    private void SetMeleeItem(TemplateItem item, JsonObject config, ModDatabaseTables databaseTables, JsonNode? rootConfig)
    {
        var slash = GetNumberValue(config["slashDamage"]);
        var stab = GetNumberValue(config["stabDamage"]);

        var props = item.Properties;
        if (props is null)
        {
            return;
        }

        if (slash != null)
        {
            props.KnifeHitSlashDam = (int)Math.Round(slash.Value);
        }

        if (stab != null)
        {
            props.KnifeHitStabDam = (int)Math.Round(stab.Value);
        }
    }
}
